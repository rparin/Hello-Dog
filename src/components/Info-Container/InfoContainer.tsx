import { cn } from "@/lib/utils";
import Infobar, { infoBarType } from "./Infobar";

export default function InfoContainer({
  className,
  title,
  info,
}: {
  className?: string;
  title: string;
  info: Array<infoBarType>;
}) {
  const barItems = info.map((item) => {
    return (
      <Infobar className="md:mb-1" key={item.name} barValue={item}></Infobar>
    );
  });

  return (
    <div
      className={cn(
        "flex h-full w-full flex-col items-center bg-accent text-black",
        className
      )}>
      <h1 className="text-base leading-none md:mb-0.5 md:text-xl">{title}</h1>

      <div className="hide-scrollbar flex h-full w-full flex-col overflow-auto">
        {barItems}
      </div>
    </div>
  );
}
