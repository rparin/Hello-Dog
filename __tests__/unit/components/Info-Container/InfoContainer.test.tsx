import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { axe, toHaveNoViolations } from "jest-axe";
import InfoContainer from "@/components/Info-Container/InfoContainer";

const PROPS = {
  title: "Physical Stats",
  info1: {
    name: "Life Expectancy",
    value: {
      other: "12 - 15 years",
    },
  },
};

test("it renders component unchanged", () => {
  const { container } = render(
    <InfoContainer title={PROPS.title} info={[PROPS.info1]}></InfoContainer>
  );
  expect(container).toMatchSnapshot();
});

test("it should show Info container title", () => {
  render(
    <InfoContainer title={PROPS.title} info={[PROPS.info1]}></InfoContainer>
  );
  const infoTitleElement = screen.getByText(PROPS.title);
  expect(infoTitleElement).toBeInTheDocument();
});

expect.extend(toHaveNoViolations);
test("should have no accessibility violations", async () => {
  const { container } = render(
    <InfoContainer title={PROPS.title} info={[PROPS.info1]}></InfoContainer>
  );
  const results = await axe(container);
  expect(results).toHaveNoViolations();
});
