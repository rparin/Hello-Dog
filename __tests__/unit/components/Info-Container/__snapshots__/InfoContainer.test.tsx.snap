// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders component unchanged 1`] = `
<div>
  <div
    class="flex h-full w-full flex-col items-center bg-accent text-black"
  >
    <h1
      class="text-base leading-none md:mb-0.5 md:text-xl"
    >
      Physical Stats
    </h1>
    <div
      class="hide-scrollbar flex h-full w-full flex-col overflow-auto"
    >
      <div
        class="flex h-full w-full flex-col md:mb-1"
      >
        <p
          class="ml-1.5 mr-1.5 border-b border-solid border-black text-xs leading-none md:text-base"
        >
          Life Expectancy
        </p>
        <div
          class="mx-1 mt-0.5 flex h-full max-h-7 items-center justify-center rounded-3xl bg-white text-xs md:mx-0.5 md:text-base"
        >
          <p
            class="leading-none"
          >
            12 - 15 years
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
